modules/masterbar/masterbar.php

```diff
@@ -0,0 +1,821 @@
+			return;
+		}
+
+		$this->user_data = Jetpack::get_connected_user_data( $this->user_id );
+		$this->user_login = $this->user_data['login'];
+		$this->display_name = $this->user_data['display_name'];
+		$this->primary_site_slug = $this->get_site_slug();
+		// We need to use user's setting here, instead of relying on current blog's text direction
+		$this->user_text_direction = $this->user_data['text_direction'];
+
+		if ( $this->is_rtl() ) {
+			// Extend core WP_Admin_Bar class in order to add rtl styles
+			add_filter( 'wp_admin_bar_class', function () {
+				return 'RTL_Admin_Bar';
+			} );
+		}
+
+		add_action( 'wp_before_admin_bar_render', array( $this, 'replace_core_masterbar' ), 99999 );
+
+		add_action( 'wp_head', array( $this, 'add_styles_and_scripts' ) );
+		add_action( 'admin_head', array( $this, 'add_styles_and_scripts' ) );
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'remove_core_styles' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'remove_core_styles' ) );
+
+		if ( Jetpack::is_module_active( 'notes' ) && $this->is_rtl() ) {
+			// Override Notification module to include RTL styles
+			add_action( 'enqueue_rtl_notification_styles', '__return_true' );
+		}
+	}
+
+	public function remove_core_styles() {
+		wp_dequeue_style( 'admin-bar' );
+	}
+
+	public function is_rtl() {
+		return $this->user_text_direction === 'rtl' ? true : false;
+	}
+
+	public function add_styles_and_scripts() {
+
+		if ( $this->is_rtl() ) {
+			wp_enqueue_style( 'a8c-wpcom-masterbar-rtl', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/rtl/wpcom-admin-bar-rtl.css' ) );
+			wp_enqueue_style( 'a8c-wpcom-masterbar-overrides-rtl', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/rtl/masterbar-rtl.css' ) );
+		} else {
+			wp_enqueue_style( 'a8c-wpcom-masterbar', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/wpcom-admin-bar.css' ) );
+			wp_enqueue_style( 'a8c-wpcom-masterbar-overrides', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/masterbar.css' ) );
+		}
+
+		// Local overrides
+		wp_enqueue_style( 'a8c_wpcom_css_override', plugins_url( 'overrides.css', __FILE__ ) );
+
+		if ( ! Jetpack::is_module_active( 'notes ' ) ) {
+			// Masterbar is relying on some icons from noticons.css
+			wp_enqueue_style( 'noticons', $this->wpcom_static_url( '/i/noticons/noticons.css' ), array(), JETPACK_NOTES__CACHE_BUSTER );
+		}
+
+		wp_enqueue_script( 'a8c_wpcom_masterbar_overrides', $this->wpcom_static_url( '/wp-content/mu-plugins/admin-bar/masterbar-overrides/masterbar.js' ) );
+	}
+
+	public function get_site_slug() {
+		$url = get_site_url();
+		$url = parse_url( $url );
+		$url = $url['host'] . untrailingslashit( $url['path'] );
+
+		return str_replace( '/', '::', $url );
+	}
+
+	function wpcom_static_url( $file ) {
+		$i   = hexdec( substr( md5( $file ), - 1 ) ) % 2;
+		$url = 'http://s' . $i . '.wp.com' . $file;
+
+		return set_url_scheme( $url );
```

Comment:
No advantage, it's just something that I've overlooked. Updated to use HTTPS in 0660ea26a58439a22995179e04b38be434d821f6.